tests/test_nhg.py

```diff
@@ -216,6 +287,358 @@ def test_route_nhg(self, dvs, dvs_route, testlog):
+        nhg_ps._del('group3')
+
+        # Group3's NH should be deleted
+        asic_db.wait_for_n_keys(self.ASIC_NHS_STR, asic_nhs_count + 2)
+
+        # Delete group2
+        nhg_ps._del('group2')
+
+        # The number of NHs should be the same as they are still referenced by
+        # group1
+        time.sleep(1)
+        asic_db.wait_for_n_keys(self.ASIC_NHS_STR, asic_nhs_count + 2)
+
+        # Update group1 with no weights and both labeled and unlabeled NHs
+        fvs = swsscommon.FieldValuePairs([('nexthop', '10.0.0.1,10.0.0.3'),
+                                          ('mpls_nh', 'push2,na'),
+                                            ('ifname', 'Ethernet0,Ethernet4')])
+        nhg_ps.set('group1', fvs)
+
+        # Group members should be replaced and one NH should get deleted
+        asic_db.wait_for_n_keys(self.ASIC_NHGM_STR, asic_nhgms_count + 2)
+        asic_db.wait_for_n_keys(self.ASIC_NHS_STR, asic_nhs_count + 1)
+
+        # Assert the weights of the NHGMs are the expected ones
+        nhgm_ids = self.get_nhgm_ids('group1', dvs)
+        weights = []
+        for nhgm_id in nhgm_ids:
+            fvs = asic_db.get_entry(self.ASIC_NHGM_STR, nhgm_id)
+            weights.append(fvs['SAI_NEXT_HOP_GROUP_MEMBER_ATTR_WEIGHT'])
+        assert weights == ['0', '0']
+
+        # Delete group1
+        nhg_ps._del('group1')
+
+        # Wait for the group and it's members to be deleted
+        asic_db.wait_for_n_keys(self.ASIC_NHG_STR, asic_nhgs_count)
+        asic_db.wait_for_n_keys(self.ASIC_NHGM_STR, asic_nhgms_count)
+
+        # The two next hops should also get deleted
+        asic_db.wait_for_n_keys(self.ASIC_NHS_STR, asic_nhs_count)
+
+    def test_nhgorch_excp_group_cases(self, dvs, testlog):
+        for i in range(3):
+            self.config_intf(i, dvs)
+
+        app_db = dvs.get_app_db()
+        asic_db = dvs.get_asic_db()
+        nhg_ps = swsscommon.ProducerStateTable(app_db.db_connection, "NEXTHOP_GROUP_TABLE")
+        rt_ps = swsscommon.ProducerStateTable(app_db.db_connection, "ROUTE_TABLE")
+
+        def get_nhg_keys():
+            return asic_db.get_keys(self.ASIC_NHG_STR)
+
+        def get_nhgm_keys():
+            return asic_db.get_keys(self.ASIC_NHGM_STR)
+
+        def get_rt_keys():
+            return asic_db.get_keys(self.ASIC_RT_STR)
+
+        # Count existing objects
+        prev_nhg_keys = get_nhg_keys()
+        asic_nhgs_count = len(get_nhg_keys())
+        asic_nhgms_count = len(get_nhgm_keys())
+        asic_nhs_count = len(asic_db.get_keys(self.ASIC_NHS_STR))
+
+        # Remove a group that does not exist
+        nhg_ps._del("group1")
+        time.sleep(1)
+        assert len(get_nhg_keys()) == asic_nhgs_count
+
+        # Create a next hop group with a member that does not exist - should fail
+        fvs = swsscommon.FieldValuePairs([('nexthop', '10.0.0.1,10.0.0.3,10.0.0.63'),
+                                        ("ifname", "Ethernet0,Ethernet4,Ethernet124")])
```

Comment:
@prsunny If one member is invalid don't we program the rest of the valid members?